CRM/Core/Payment.php

```diff
@@ -1288,14 +1288,23 @@ public static function paypalRedirect(&$paymentProcessor) {
    * Page callback for civicrm/payment/ipn
    */
   public static function handleIPN() {
-    self::handlePaymentMethod(
-      'PaymentNotification',
-      array(
-        'processor_name' => @$_GET['processor_name'],
-        'processor_id' => @$_GET['processor_id'],
-        'mode' => @$_GET['mode'],
-      )
-    );
+    try {
+      self::handlePaymentMethod(
+        'PaymentNotification',
+        [
+          'processor_name' => CRM_Utils_Request::retrieveValue('processor_name', 'String'),
+          'processor_id' => CRM_Utils_Request::retrieveValue('processor_id', 'Integer'),
+          'mode' => CRM_Utils_Request::retrieveValue('mode', 'Alphanumeric'),
+        ]
+      );
+    }
+    catch (CRM_Core_Exception $e) {
+      Civi::log()->error('payment_callback_exception', [
```

Comment:
Can we have "ipn" somewhere in the name?  Makes tracing the source a bit easier when found in logs

Reply:
I've added it - I was  a little loath as ipn feels like a paypal term rather than a generic term but I guess we do tend to use it generically

Reply:
Great :-) Maybe this: https://bitpay.com/docs/invoice-callbacks or https://stripe.com/docs/charges makes you feel more comfortable about the term IPN?